src/main/java/com/internet/shop/dao/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,232 @@
+                    + user.getLogin(), e);
+        }
+        setIdToRoles(user.getRoles());
+        addRolesToUser(user);
+        return user;
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        User user;
+        Set<Role> roles = new HashSet<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM users "
+                    + " JOIN user_roles ON users.user_id = user_roles.user_id"
+                    + " JOIN roles ON user_roles.role_id = roles.role_id"
+                    + " WHERE users.user_id = ? AND deleted = false";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                user = setUserFields(resultSet);
+                roles.add(getRolesWithId(resultSet));
+            } else {
+                return Optional.empty();
+            }
+            while (resultSet.next()) {
+                roles.add(getRolesWithId(resultSet));
+                user.setRoles(roles);
+            }
+            user.setRoles(roles);
+            return Optional.of(user);
+        } catch (SQLException e) {
+            throw new DataProcessingException("User by id " + id
+                    + "is not found", e);
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        List<User> users = new ArrayList<>();
+        String query = "SELECT * FROM users WHERE deleted = false";
+        try (PreparedStatement preparedStatement =
+                     ConnectionUtil.getConnection().prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                User user = setUserFields(resultSet);
+                users.add(user);
+            }
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+        users.forEach(this::getUserRoles);
+        return users;
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE users "
+                + "SET user_name = ?, user_login = ?, user_pass = ? "
+                + "WHERE user_id = ?";
+        try (PreparedStatement preparedStatement =
+                     ConnectionUtil.getConnection().prepareStatement(query)) {
+            preparedStatement.setString(1, user.getName());
+            preparedStatement.setString(2, user.getLogin());
+            preparedStatement.setString(3, user.getPassword());
+            preparedStatement.setLong(4, user.getId());
+            preparedStatement.executeQuery();
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+        deleteUserRoles(user.getId());
+        setIdToRoles(user.getRoles());
```

Comment:
in update method you should receive roles with ids, so this line seems to be redundant

Reply:
fixed query in update method delete redundant lines